### ISSUE TITLE
[BUG] Environment check fails NFS test if /boot/config-* does not exist

### ISSUE THREAD

**Issue by kernle32dll** [longhorn/longhorn#5679]
## Describe the bug (🐛 if you encounter this issue)

I run a cluster setup, where there are no `/boot/config-*` files. This leads to the environment check script failing, since it can't check for `CONFIG_NFS_V4_1` and `CONFIG_NFS_V4_2`. 

## To Reproduce

1. Setup a cluster, which does not have `/boot/config-*` files (for me, a manual Arch Linux install, but a Pi could probably work, too)
2. Execute the environment check
3. See error

## Expected behavior
No errors

## Additional context

Judging by [this](https://superuser.com/a/287372), I guess `/proc/config.gz` would be a better location to check. It seems to work on my system at least.

**Comment by mantissahz** [comment]
@kernle32dll Thanks for reporting the issue.
Here are [operating systems](https://longhorn.io/docs/1.4.1/best-practices/#operating-system) we supported now, but it is good to improve the script to support more operating systems.
Thanks again.

**Comment by kernle32dll** [comment]
> @kernle32dll Thanks for reporting the issue. Here are [operating systems](https://longhorn.io/docs/1.4.1/best-practices/#operating-system) we supported now, but it is good to improve the script to support more operating systems. Thanks again.

Fair point. I was not aware of this, as the check script actually has some Arch related code (pacman).

**Comment by denniseffing** [comment]
Just ran into the same issue. @kernle32dll thanks for pointing me in the right direction with `/proc/config.gz`.

**Comment by derekbit** [comment]
@mantissahz Let's address it.

**Comment by COLDTURNIP** [comment]
Currently, Longhorn check NFS kernel capability in two ways.

- Longhorn manager: checks `/boot/config-*` and `/proc/config.gz`
- CLI check preflight: checks only `/boot/config-*`

However, Arch Linux provides only `/proc/config.gz`. We should sync the logic from manager to CLI checker.

Related ticket: longhorn/longhorn#9830
